File: app/src/main/java/com/apache/fastandroid/jetpack/lifecycle/LifecycleHandler.java
package com.apache.fastandroid.jetpack.lifecycle;


import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.LifecycleObserver;
import android.arch.lifecycle.LifecycleOwner;
import android.arch.lifecycle.OnLifecycleEvent;
import android.os.Handler;
import android.os.Looper;

import com.apache.fastandroid.util.MainLogUtil;

/**
 * Created by Jerry on 2020-05-16.
 */
public class LifecycleHandler extends Handler implements LifecycleObserver {

    private LifecycleOwner lifecycleOwner;

    public LifecycleHandler(final LifecycleOwner lifecycleOwner) {
        this.lifecycleOwner = lifecycleOwner;
        addObserver();
    }

    public LifecycleHandler(final Callback callback, final LifecycleOwner lifecycleOwner) {
        super(callback);
        this.lifecycleOwner = lifecycleOwner;
        addObserver();
    }

    public LifecycleHandler(final Looper looper, final LifecycleOwner lifecycleOwner) {
        super(looper);
        this.lifecycleOwner = lifecycleOwner;
        addObserver();
    }

    public LifecycleHandler(final Looper looper, final Callback callback, final LifecycleOwner lifecycleOwner) {
        super(looper, callback);
        this.lifecycleOwner = lifecycleOwner;
        addObserver();
    }

    private void addObserver() {
        lifecycleOwner.getLifecycle().addObserver(this);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    private void onDestroy() {
        MainLogUtil.d("onDestroy removeCallbacksAndMessages");
        removeCallbacksAndMessages(null);
        lifecycleOwner.getLifecycle().removeObserver(this);
    }


}

File: fastFramework/src/main/java/com/tesla/framework/component/lifecycle/LifecycleHandler.kt
package com.tesla.framework.component.lifecycle

import android.os.Handler
import android.os.Looper
import androidx.lifecycle.DefaultLifecycleObserver
import androidx.lifecycle.LifecycleOwner
import com.tesla.framework.common.util.Preconditions
import com.tesla.framework.component.lifecycle.LifecycleHandler
import com.tesla.framework.component.logger.Logger

class LifecycleHandler(val lifecycleOwner: LifecycleOwner,looper: Looper = Looper.getMainLooper()) : Handler(looper), DefaultLifecycleObserver {

    init {
        lifecycleOwner.lifecycle.addObserver(this)
    }

    override fun onDestroy(owner: LifecycleOwner) {

        lifecycleOwner.lifecycle.removeObserver(this)
        removeCallbacksAndMessages(null)
    }

}